pylint/checkers/deprecated.py

```diff
@@ -59,6 +59,11 @@ class DeprecatedMixin(BaseChecker):
         ),
     }
 
+    def check_consistency(self):
+        """This is a mixin, so there is no expectation that the messages should be
+        mutually exclusive. They will overlap between checkers by design.
+        """
```

Comment:
What's the problem here ? I think the msgid could still be consistent i.e. they should all start by "15" and be valid msgid with a letter and 4 number ?

Reply:
Both `StdlibChecker` and `ImportsChecker` emit `W0402`, `W0404`... etc for `deprecated-module` and friends. This raises `InvalidMessageError`:

```
Inconsistent checker part in message id 'W0402' (expected 'x15xx' ...
```
Are they supposed to become W1502 and W1504 in the stdlib checker? That would defeat the point of the Mixin, I guess? I don't know what the backward-compatibility issues would be.

Reply:
Well, we have more uncomfortable problems because now we have the docs complaining about two messages. I guess the recent infrastructure assumed no checkers shared messages.
```
/home/runner/work/pylint/pylint/doc/messages/messages_list.rst:185: WARNING: duplicated entry found in toctree: messages/warning/deprecated-module
```

Reply:
Oh. The only overlap is W0402 `deprecated-module`. They differ in this way: The ImportsChecker looks at modules given in the config. The StdlibChecker looks at the hard-coded dictionary we maintain.

Reply:
Maybe the answer is to just let the `ImportsChecker` check this message and check both sources. Currently if `optparse` is given at the command line, we emit twice, once by each checker:
```
pylint/config/option_manager_mixin.py:11:0: W0402: Uses of a deprecated module 'optparse' (deprecated-module)
pylint/config/option_manager_mixin.py:11:0: W0402: Uses of a deprecated module 'optparse' (deprecated-module)
```

Reply:
Decided to go that route in [1e6bdcf]

Reply:
Sorry I was not able to answer sooner. I think one of the possibility offered by the mixin was to be able to extends it easily see https://github.com/PyCQA/pylint/pull/4225. I'm not sure this will still be possible if we move it to the import checker ?

Reply:
No worries. I think it's not a problem, since I didn't end up touching `DeprecatedMixin` after all. The only change is for `StdlibChecker` to stop checking for deprecated modules, and for `ImportsChecker` to add the stdlib modules to the user-given modules. A plugin seeing a behavior change would be a plugin inheriting from one of the two checkers, not the mixin, so it's just like any behavior change.